Added Docstrings and Functions

thevoxium opened the pull request.

thevoxium committed:
first for issue #6

thevoxium committed:
added docstring to functions

americast commented on kmain.py:
```diff
@@ -151,127 +219,6 @@ def categorical_accuracy_mod(y_true, y_pred):
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-
-  np.save("out_1_lrp", analysis)
```
Hi @thevoxium! Thanks a lot for the PR. May I know why you have removed these lines?

thevoxium replied:
I was just experimenting with that can I implement it using functions. I will change it in the next commit.

americast replied:
Yes, it would be great if you can implement it using a function. Moreover, you could perhaps make a list containing the no of layers you wish to have, and have a loop iterating over that which keeps calling the LRP function.

thevoxium replied:
@americast I asked a doubt regarding LRP on gitter. It would be great if you look at please.

americast commented on kmain.py:
```diff
@@ -76,32 +160,16 @@
+print("Would you like to load previous saved model (y/n): ")
+model_choice = input()
+if (model_choice=='y' or model_choice=='Y'):
+  model = load_model("model.h5")
```
Is this not inconsistent with [Line No 189](https://github.com/americast/algo_bias/pull/13/files#diff-0d2eb74693a923211c75e8d685eb25f5a6e02866f3598b5499d426f5eebd1701R189)?

thevoxium replied:
I forgot to change that, thanks for the reminder.

thevoxium committed:
function for LRP

thevoxium commented:
Please review this.

thevoxium committed:
added GIF of LRP heatmaps

thevoxium committed:
added GIF in readme.md

thevoxium commented:
-Added GIF of heatmaps.
-Added Functions to `kmain.py`.
-Added Docstrings of those functions.
-Changed `lrp_matrix.py` and `plot.py` accordingly. 

**Please review this @americast and @rakeshbal99 .**

thevoxium committed:
reversed to previous commit

americast commented on kmain.py:
```diff
@@ -12,11 +12,96 @@
+  Args:
+      model: The neural network to perform analysis upon.
+  """
+  layer_num = [2,6,10,14,18,22,26,30,34,38,42, 44]
```
Can we have this as an input to the `layer_analysis` function?

americast reviewed (changes requested):
@thevoxium Great! Thanks a lot! I have requested a small change. It would be great if you can kindly incorporate the same.

thevoxium committed:
layer_num passed to func

americast reviewed (approved):
Looks good! Thanks @thevoxium!
